github/release_notes/model.py

```diff
@@ -83,17 +85,29 @@ def __init__(
         reference_type: ReferenceType,
         reference_id: str,
         user_login: str,
-        source_repo: str,
-        cn_current_repo: ComponentName,
+        current_component: gci.componentmodel.Component,
+        source_component: gci.componentmodel.Component,
     ):
         if reference_id:
-            check_type(reference_id, str)
+            ci.util.check_type(reference_id, str)
+
+        ci.util.not_none(source_component)
 
         reference = Reference(type=reference_type, identifier=reference_id)
 
-        cn_source_repo = ComponentName(name=source_repo)
-        is_current_repo = cn_current_repo == cn_source_repo
-        from_same_github_instance = cn_current_repo.github_host() == cn_source_repo.github_host()
+        source_component_access = cnudie.util.determine_main_source_for_component(
+            source_component
+        ).access
+        current_component_access = cnudie.util.determine_main_source_for_component(
+            current_component
+        ).access
```

Comment:
you should also check that it is a github-access, please (I know, currently this is the only one. however, cnudie allows different access-types)